packages/espressocash_app/lib/features/kyc_sharing/screens/basic_information_screen.dart

```diff
@@ -0,0 +1,231 @@
+import 'package:flutter/material.dart';
+import 'package:intl/intl.dart';
+
+import '../../../di.dart';
+import '../../../l10n/l10n.dart';
+import '../../../ui/bottom_button.dart';
+
+import '../../../ui/colors.dart';
+import '../../../ui/loader.dart';
+import '../../../ui/radio_button.dart';
+import '../../../ui/snackbar.dart';
+import '../../country_picker/models/country.dart';
+import '../../country_picker/widgets/country_picker.dart';
+import '../models/document_type.dart';
+import '../services/kyc_service.dart';
+import '../widgets/document_picker.dart';
+import '../widgets/kyc_page.dart';
+import '../widgets/kyc_text_field.dart';
+
+class BasicInformationScreen extends StatefulWidget {
+  const BasicInformationScreen({super.key});
+
+  static Future<bool> push(BuildContext context) => Navigator.of(context)
+      .push<bool>(
+        MaterialPageRoute(
+          builder: (context) => const BasicInformationScreen(),
+        ),
+      )
+      .then((result) => result ?? false);
+
+  @override
+  State<BasicInformationScreen> createState() => _BasicInformationScreenState();
+}
+
+class _BasicInformationScreenState extends State<BasicInformationScreen> {
+  final _idController = TextEditingController();
+  final _firstNameController = TextEditingController();
+  final _middleNameController = TextEditingController();
+  final _lastNameController = TextEditingController();
+  final _dobController = TextEditingController();
+
+  bool _isShareData = false;
+
+  DateTime? _dob;
+
+  Country? _country;
+  DocumentType? _idType;
+
+  bool get _isValid =>
+      _firstNameController.text.isNotEmpty &&
+      _lastNameController.text.isNotEmpty &&
+      _dobController.text.isNotEmpty &&
+      _idController.text.isNotEmpty &&
+      _isShareData &&
+      _idType != null &&
+      _country != null;
+
+  Future<void> _handleSubmitted() async {
+    final success = await runWithLoader<bool>(
+      context,
+      () async {
+        try {
+          final countryCode = _country?.code;
+          final idTypeValue = _idType?.value;
+
+          if (countryCode == null || idTypeValue == null) {
+            throw Exception();
+          }
+
+          await sl<KycSharingService>().updateBasicInfo(
+            firstName: _firstNameController.text,
+            lastName: _lastNameController.text,
+            dob: _dob,
+            countryCode: countryCode,
+            idType: _idType,
+            idNumber: _idController.text,
+          );
+
+          if (!mounted) return false;
+
+          return true;
+        } on Exception {
+          if (!mounted) return false;
+
+          showCpErrorSnackbar(
+            context,
+            message: context.l10n.failedToUpdateData,
+          );
+
+          return false;
+        }
+      },
+    );
+    if (!mounted) return;
+    if (success) Navigator.pop(context, true);
+  }
+
+  Future<void> _selectDob() async {
+    final DateTime? picked = await showDatePicker(
```

Comment:
IIRC, we discussed that. It shouldn't be calendar view, just an input field with proper formatting. Calendar is not convenient for selecting known date, such as date of birth.

Reply:
updated to use textfield